File: Python/0150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operands = ['+', '-', '*', '/']
        for token in tokens:
            if token not in operands:
                stack.append(int(token))
            else:
                temp = int(eval(f'{stack[-2]}{token}{stack[-1]}'))
                del stack[-2:]
                stack.append(temp)
        return stack[-1]

File: C++/0150.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int64_t> stack;
        set<string> operands = {"+", "-", "*", "/"};

        for (const auto& ch : tokens) {
            if (!operands.count(ch)) {
                stack.push(stoi(ch));
            } else {
                int64_t second = stack.top();
                stack.pop();
                int64_t first = stack.top();
                stack.pop();

                switch (ch[0]) {
                    case '+':
                        stack.push(first + second);
                        break;
                    case '-':
                        stack.push(first - second);
                        break;
                    case '*':
                        stack.push(first * second);
                        break;
                    case '/':
                        stack.push(first / second);
                        break;
                }
            }
        }

        return stack.top();
    }
};
